File: typescript/src/models/messageInfo.ts
// this file is @generated
import { type ContentType, ContentTypeSerializer } from "./contentType";

export interface MessageInfo {
  param?: string | null;
  contentType: ContentType;
  content: string;
}

export const MessageInfoSerializer = {
  _fromJsonObject(object: any): MessageInfo {
    return {
      param: object["param"],
      contentType: ContentTypeSerializer._fromJsonObject(object["content_type"]),
      content: object["content"],
    };
  },

  _toJsonObject(self: MessageInfo): any {
    return {
      param: self.param,
      content_type: ContentTypeSerializer._toJsonObject(self.contentType),
      content: self.content,
    };
  },
};

File: rust/src/models/message_info.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::content_type::ContentType;

#[derive(Clone, Debug, PartialEq, Deserialize, Serialize)]
pub struct MessageInfo {
    #[serde(skip_serializing_if = "Option::is_none")]
    pub param: Option<String>,
    pub content_type: ContentType,
    pub content: String,
}
